source/common/ssl/ssl_socket.cc

```diff
@@ -301,6 +319,32 @@ std::string SslSocket::getUriSanFromCertificate(X509* cert) {
   return result;
 }
 
+std::vector<std::string> SslSocket::getDnsSansFromCertificate(X509* cert) {
```

Comment:
You can use ranged for loop here instead (see https://github.com/envoyproxy/envoy/pull/3081).